### ISSUE TITLE
tracer utility does not support verifyta trace format

### ISSUE THREAD

**Issue by pacalet** [UPPAALModelChecker/utap#1]
Under Debian 11.1 (bullseye) there is an inconsistency between the `verifyta` utility that comes with UPPAAL 4.1.25-5 and the `tracer` utility that comes with `libutap` 0.94. Tools versions:

```bash
$ verifyta --version
(Academic) UPPAAL 4.1.25-5 (rev. 643E9477AA51E17F), April 2021
Copyright (c) 1995 - 2021, Uppsala University and Aalborg University.
All rights reserved.
Compiled with g++-10 -Wall -Wempty-body -DNDEBUG -O3 -ffloat-store -fPIC -mtune=generic -DLIBXML_STATIC -DENABLE_PROB -DENABLE_BRANCHING -DENABLE_BRANCHING -DENABLE_LSC -DENABLE_64BIT -DLOCSTORE -DENABLE_POLYHEDRA_CHECKING  -DBOOST_DISABLE_THREADS -Wsuggest-override
Free for academic and non-commercial use only.
$ head -1 /opt/libutap/share/doc/libutap/README
libutap, version 0.94
```

To reproduce the bug first download the [XML file of the example system](https://github.com/UPPAALModelChecker/utap/files/7633986/sys.txt) and rename it as `sys.xml`, then:

```bash
$ verifyta -t0 -fsys sys.xml
Options for the verification:
  Generating some trace
  Search order is breadth first
  Using conservative space optimisation
  Seed is 1638362312
  State space representation uses minimal constraint systems

Verifying formula 1 at /nta/queries/query[1]/formula
 -- Formula is NOT satisfied.
Writing counter example to sys-1.xtr
$ UPPAAL_COMPILE_ONLY=1 verifyta sys.xml - > sys.if
$ tracer sys.if sys-1.xtr
State: Expecting a line with '.' but got ' '
```

The issue seems to come from trailing spaces at the end of some lines of the `sys-1.xtr` file generated by `verifyta`:

```bash
$ cat -E sys-1.xtr
0 $
.$
.$
$
.$
.$
```

A possible fix consists in using `boost` to trim the leading and trailing spaces in `tracer.cpp`:

```bash
$ git diff
diff --git a/src/tracer.cpp b/src/tracer.cpp
index f0a4274..4c82e22 100644
--- a/src/tracer.cpp
+++ b/src/tracer.cpp
@@ -30,6 +30,7 @@
 #include <stdexcept>
 #include <string>
 #include <vector>
+#include <boost/algorithm/string/trim.hpp>
 
 /* This utility takes an UPPAAL model in the UPPAAL intermediate
  * format and a UPPAAL XTR trace file and prints trace to stdout in a
@@ -180,10 +181,11 @@ istream& readdot(istream& is)
 {
     string str;
     getline(is, str);
-    if (str.empty())
+    while (str.empty())
     {
         getline(is, str);
     }
+    boost::algorithm::trim(str);
     if (str != ".")
     {
         cerr << "Expecting a line with '.' but got '" << str << "'" << endl;
```

**Comment by mikucionisaau** [comment]
Thanks for the patch, @pacalet! I think boost is too much overhead here, so I did it my way.
Could you please try the `robust-against-empty-lines` branch?
like:
```sh
git fetch
git checkout robust-against-empty-lines
```
then recompile (using `make`) and try your case?

**Comment by mikucionisaau** [comment]
I believe this is solved, merged and released as 0.94. 
Please reopen in case it persists.
